### PR TITLE
CSHARP-1000 Fix support for astra custom domain

### PR DESCRIPTION
**Pull request by joao-r-reis**

### CODE REVIEW

File: src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs [hunk 1]
```diff
@@ -145,6 +148,55 @@ public bool Validate(X509Certificate cert, X509Chain chain, SslPolicyErrors erro
...
+
+            return false;
+        }
+
+        private IEnumerable<string> GetSubjectAlternativeNames(X509Certificate2 cert)
+        {
+            var result = new List<string>();
+
+            var subjectAlternativeName = cert.Extensions.Cast<X509Extension>()
+                .Where(n => n.Oid.Value == "2.5.29.17") //n.Oid.FriendlyName=="Subject Alternative Name")
```
Comment (absurdfarce) [inline src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs:177]: Perhaps use the SubjectAlternateNameOid constant. you defined above here instead?

File: src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs [hunk 2]
```diff
@@ -145,6 +148,55 @@ public bool Validate(X509Certificate cert, X509Chain chain, SslPolicyErrors erro
...
+                .Select(n => n.Format(true))
+                .FirstOrDefault();
+
+            if (subjectAlternativeName != null)
+            {
+                var alternativeNames = subjectAlternativeName.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
+
+                foreach (var alternativeName in alternativeNames)
+                {
+                    var groups = Regex.Match(alternativeName, @"^(.*)=(.*)").Groups; // @"^DNS Name=(.*)").Groups;
```
Comment (absurdfarce) [inline src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs:188]: Is this intended to... replace the DnsNameRegex referenced above?  Seems like it's trying to do the same thing with a different regex.

**Review by absurdfarce — APPROVED** [review]
Certainly seems to address the problem as described.  A few nits based on my understanding of the code but I don't think either of them are problematic enough to hold things up if you need to get a release out.

It might be nice to have a test to confirm that SANs were being used for validation but, again, I don't think it's essential.

File: src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs (hunk 2 above)
Comment (joao-r-reis) [reply src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs:187]: Ahh I had to try multiple solutions and forgot about those constants... fixed

File: src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs (hunk 1 above)
Comment (joao-r-reis) [reply src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs]: Ahh I had to try multiple solutions and forgot about those constants... fixed

**Comment by absurdfarce** [comment]
👍 to the most recent round of changes.  Nice tests @joao-r-reis !

**Pull request merged**